### ISSUE TITLE
Validation bugs in the new notebook form

### ISSUE THREAD

**Issue by mathis-marcotte** [StatCan/jupyter-apis#189]
While testing, I noticed some of the validation in the form for creating a new notebook were not working properly.

- The volume names have server-side validation , where it must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character. But this validation isn't on the client-side. I think it could be done just like with the notebook name error that gets displayed when it doesn't follow the validation. This is the regex returned by the back-end for the validation of the volume names : `[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')`
- The requested CPU and RAM inputs are text inputs, even though they are expecting number values. These could probably be changed to follow the design of the CPU and RAM limit inputs.
- It seems like for some errors, the requested cpu and the cpu limit inputs share the same error property and message even when only one of these fields is at fault. This also happens for the requested RAM and ram limit inputs.
- When inputing 0 for the requested cpu input, it says that the minimum value is 0.1. But then, if you input a value smaller than 0.5, it says that the minimum is 0.5. This should be fixed to always display the correct minimum value(0.5)

**Comment by mathis-marcotte** [comment]
Here are before and after snippets of the behavior of the error messages for the cpu and ram inputs. You can see that in the first picture, the messages weren't always disaplyed correctly before they would be shared by both the requested and limit inputs.

before: 
![image](https://user-images.githubusercontent.com/84033116/206216897-061ef40e-ff8d-4c8e-a21a-a38052797ac9.png)

after:
![image](https://user-images.githubusercontent.com/84033116/206217032-acf9d8c9-835c-414a-a79c-d6cb868be700.png)

**Comment by mathis-marcotte** [comment]
For the first point about volume name validation not being present on the client-side, after some investigation, I noticed there was in fact validators already in place for this. The issue was only happening with workspace volumes when you opened a fresh new form for creating a new notebook. For some reason, the default workspace volume that is already created in a new form was getting it's name validators overwritten to only have the "Required" validator, therefore losing that Pattern validator. So I added a fix to make sure the correct validators get reapplied.
